src/bio/bio_monitor.c

```diff
@@ -655,18 +928,81 @@ bio_init_health_monitoring(struct bio_blobstore *bb, char *bdev_name)
 	BIO_PROTO_NVME_STATS_LIST
 #undef X
 	D_FREE(binfo);
+}
 
-	return 0;
+/*
+ * Register DAOS metrics to export Intel Vendor SMART NVMe SSD attributes.
+ */
+void
+bio_export_vendor_health_stats(struct bio_blobstore *bb, char *bdev_name)
+{
+	struct bio_dev_info		*binfo;
+	int				 rc;
 
-free_error_buf:
-	spdk_dma_free(bb->bb_dev_health.bdh_error_buf);
-	bb->bb_dev_health.bdh_error_buf = NULL;
-free_ctrlr_buf:
-	spdk_dma_free(bb->bb_dev_health.bdh_ctrlr_buf);
-	bb->bb_dev_health.bdh_ctrlr_buf = NULL;
-free_health_buf:
-	spdk_dma_free(bb->bb_dev_health.bdh_health_buf);
-	bb->bb_dev_health.bdh_health_buf = NULL;
+	D_ALLOC_PTR(binfo);
+	if (binfo == NULL) {
+		D_WARN("Failed to allocate binfo\n");
+		return;
+	}
+#define Y(field, fname, desc, unit, type)				\
```

Comment:
(style)  Macros with multiple statements should be enclosed in a do - while loop